core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -45,8 +51,10 @@ case class SatoshisPerKiloByte(currencyUnit: CurrencyUnit)
         throw new RuntimeException(
           s"Failed to convert sat/kb -> sat/byte for ${currencyUnit}")
     }
-
   }
+
+  override def calc(tx: Transaction): CurrencyUnit =
+    Satoshis((tx.byteSize * toLong / 1000))
```

Comment:
Same as above, what's the behavior with precision here? Is this what we want?

Reply:
As an example: 

```scala
scala> 101 * 100 / 1000
val res3: Int = 10

scala> 101 * 100 * 0.001
val res4: Double = 10.1
```

Your implementation silently rounds the `0.1`, which is scary. My implementation presreves the `0.1` which would end up in a runtime exception when passing it to `Satoshis()`

Reply:
I think for the `toSatPerByte` it should be changed back, however, for the others we don't want it throw a runtime exception when calculating a fee for anything that isn't a multiple of 1000

Reply:
Curious on @nkohen's thoughts as he is the one who prompted me to make this change

Reply:
The general rule of thumb that I think we should follow is: 

Implicitly rounding decimal places in money handling libraries is a BAD idea. Be explicit and throw and make the end user _explicitly_ handle that case. This may mean we need to offer `RoundingMode` in the API.

Perhaps we should look to see how bitcoin core does this?

Reply:
![image](https://user-images.githubusercontent.com/15256660/82679732-f0b49e00-9c10-11ea-9233-0e394aa990ba.png)

Reply:
https://github.com/bitcoin/bitcoin/blob/master/src/policy/feerate.cpp#L23